server/src/main/java/com/datastrato/gravitino/server/web/rest/CatalogOperations.java

```diff
@@ -57,7 +59,12 @@ public Response listCatalogs(@PathParam("metalake") String metalake) {
           httpRequest,
           () -> {
             Namespace catalogNS = Namespace.ofCatalog(metalake);
-            NameIdentifier[] idents = manager.listCatalogs(catalogNS);
+            // Lock the root and the metalake with WRITE lock to ensure the consistency of the list.
+            NameIdentifier[] idents =
+                TreeLockUtils.doWithTreeLock(
+                    NameIdentifier.of(metalake),
+                    LockType.WRITE,
```

Comment:
Why do we use write lock, I think read lock is enough. Otherwise, it will block all the catalog operations, like load, alter.

Reply:
Let me see.

Reply:
Write lock can explicitly avoid the scenario `list` schemas and `add`, `remove`, or `alter` schemas under a specific catalog.

Reply:
Here is for catalog operations, what's the relationship with schema operations?

Reply:
It is not necessary to get parent write lock for list operations, it is too heavy to block everything. What I think is:

list -> parent read lock
create -> parent write lock
load -> itself read lock
alter -> itself write lock
drop -> parent write lock

Reply:
I agree, on the whole, as alter may change the name of an entity, I think locking itself with a write lock is not enough if we use the read lock of its parents in the list operation.

Reply:
> I agree, on the whole, as alter may change the name of an entity, I think locking itself with a write lock is not enough if we use the read lock of its parents in the list operation.

I agree.

Reply:
Solution one(Your suggestion):
- list -> parent read lock
- create -> parent write lock
- load -> itself read lock
- alter -> parent write lock
- drop -> parent write lock

Solution two: 
- list -> parent write lock
- create -> itself write lock.
- load -> itself read lock
- alter -> itself write lock
- drop -> itself write lock

I think the percentage of operations `drop`, `alter`, `create` is higher than that of `list` in general, so I prefer solution two. 
@jerryshao

Reply:
Ok, let's try solution two and do more tests to see if it is workable.